beacon-chain/blockchain/receive_attestation.go

```diff
@@ -202,10 +204,14 @@ func (s *Service) notifyEngineIfChangedHead(ctx context.Context, newHeadRoot [32
 	_, err = s.notifyForkchoiceUpdate(s.ctx, arg)
 	if err != nil {
 		log.WithError(err).Error("could not notify forkchoice update")
+		if IsInvalidBlock(err) {
+			return err
```

Comment:
I feel like either we need to propagate the error here, or log and stop propagation within `notifyForkchoiceUpdate`, doing it middle way seems like it will lead to bugs. What would be the problem in propagating here? or what would be the problem on only returning errors from `notifyForkchoiceUpdate` when `IsInvalidBlock(err) == true`?